src/components/layouts/collection-page-layout.tsx

```diff
@@ -530,24 +542,45 @@ const CollectionPageLayout: React.FunctionComponent<CoursePageLayoutProps> = ({
                     }}
                   >
                     <div
-                      className="text-gray-600 dark:text-gray-300 flex flex-row items-center rounded hover:bg-gray-200 
-                      dark:hover:bg-gray-700 border-gray-300 bg-white dark:bg-gray-800 dark:border-gray-600 px-4 py-2 border transition-colors text-sm xs:text-base ease-in-out"
+                      className={
+                        ' flex flex-row items-center rounded  px-4 py-2 border transition-all text-sm xs:text-base ease-in-out duration-150 shadow-sm ' +
+                        (isFavorite
+                          ? 'hover:bg-blue-500 bg-blue-600 border-blue-700 text-white'
+                          : 'bg-white text-gray-600 hover:bg-gray-200 dark:text-gray-300 dark:hover:bg-gray-700 border-gray-300  dark:bg-gray-800 dark:border-gray-600')
+                      }
                     >
                       <BookmarkIcon
-                        className="w-4 h-4 mr-1"
+                        className="w-4 h-4 mr-2"
                         fill={isFavorite}
                       />{' '}
-                      Bookmark
+                      {isFavorite ? 'Bookmarked' : 'Bookmark'}
                     </div>
                   </button>
                 ) : (
-                  <div
-                    className="text-gray-600 dark:text-gray-300 flex flex-row items-center rounded hover:bg-gray-200 
-                  dark:hover:bg-gray-700 border-gray-300 bg-white dark:bg-gray-800 dark:border-gray-600 px-4 py-2 border transition-colors text-sm xs:text-base ease-in-out opacity-30"
+                  <DialogButton
+                    buttonText="Bookmark"
+                    title="Sign in (or up) to bookmark"
+                    buttonStyles="text-gray-600 dark:text-gray-300 flex flex-row items-center rounded hover:bg-gray-100 
+                      dark:hover:bg-gray-700 border-gray-300 bg-white dark:bg-gray-800 dark:border-gray-600 px-4 py-2 border transition-colors text-sm xs:text-base ease-in-out opacity-90 shadow-sm"
                   >
-                    <BookmarkIcon className="w-4 h-4 mr-1" /> Bookmark
-                  </div>
+                    <LoginForm
```

Comment:
Design for the various form states needs adjusting. LoginForm component hardcodes `h2`s